wolfssl/src/ssl.rs

```diff
@@ -1041,6 +1056,26 @@ impl<IOCB: IOCallbacks> Session<IOCB> {
             e => unreachable!("{e:?}"),
         }
     }
+
+    ///  Invoked [`wolfSSL_UseKeyShare`][0]
+    ///
+    /// [0]: https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_usekeyshare
+    fn use_key_share_curve(&self, curve: CurveGroup) -> Result<()> {
+        // SAFETY: [`wolfSSL_UseKeyShare`][0] ([also][1]) expects a valid pointer to `WOLFSSL`. Per the
+        // [Library design][2] access is synchronized via the containing [`Mutex`]
+        //
+        // [0]: https://www.wolfssl.com/documentation/manuals/wolfssl/ssl_8h.html#function-wolfssl_usekeyshare
+        // [1]: https://www.wolfssl.com/doxygen/group__Setup.html#gac2d00ac65513f10e0ccd1b67d9a99e3d
+        // [2]: https://www.wolfssl.com/documentation/manuals/wolfssl/chapter09.html#thread-safety
+        match unsafe {
+            let ssl = self.ssl.lock();
+            wolfssl_sys::wolfSSL_UseKeyShare(ssl.as_ptr(), curve.as_ffi() as wolfssl_sys::word16)
+        } {
+            wolfssl_sys::WOLFSSL_SUCCESS => Ok(()),
+            x @ wolfssl_sys::WOLFSSL_FAILURE => Err(Error::fatal(self.get_error(x))),
```

Comment:
Documentation mentions about other error codes `BAD_FUNC_ARG` and `MEMORY_E` and it did not mention about `WOLFSSL_FAILURE`
Copy paste error or intentional change ?